include/pops/model.hpp

```diff
@@ -114,9 +82,44 @@ class Model
         Treatments<IntegerRaster, FloatRaster>& treatments,
         IntegerRaster& resistant,
         std::vector<std::tuple<int, int>>& outside_dispersers,  // out
-        SpreadRate<IntegerRaster>& spread_rate  // out
-    )
+        SpreadRate<IntegerRaster>& spread_rate,  // out
+        std::vector<std::vector<int>> movements)
```

Comment:
Just a detail: It would be better if the parameter is a const reference (`const std::vector<std::vector<int>>&`) like in `Simulation::movement()`. We don't need a copy here and we don't modify it as output either. Even if there would be no performance difference with the copy, cost ref better documents the intention here.